<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DatabaseOpenWidget</class>
 <widget class="QWidget" name="DatabaseOpenWidget">
  <property name="accessibleName">
   <string>Unlock KeePassXC Database</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="0,1,0,2">
   <item>
    <widget class="MessageWidget" name="messageWidget" native="true"/>
   </item>
   <item>
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::MinimumExpanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>5</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1,0">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="sizeConstraint">
      <enum>QLayout::SetDefaultConstraint</enum>
     </property>
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::MinimumExpanding</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QWidget" name="formContainer" native="true">
       <property name="maximumSize">
        <size>
         <width>700</width>
         <height>16777215</height>
        </size>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="sizeConstraint">
         <enum>QLayout::SetMinimumSize</enum>
        </property>
        <item>
         <widget class="QLabel" name="labelHeadline">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Unlock KeePassXC Database</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="ElidedLabel" name="fileNameLabel">
          <property name="text">
           <string notr="true">filename.kdbx</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>439</width>
            <height>13</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QFrame" name="horizontalFrame">
          <property name="minimumSize">
           <size>
            <width>550</width>
            <height>0</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="lineWidth">
           <number>2</number>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="sizeConstraint">
            <enum>QLayout::SetMinimumSize</enum>
           </property>
           <property name="leftMargin">
            <number>20</number>
           </property>
           <property name="topMargin">
            <number>15</number>
           </property>
           <property name="rightMargin">
            <number>20</number>
           </property>
           <property name="bottomMargin">
            <number>15</number>
           </property>
           <item>
            <widget class="QFrame" name="verticalFrame">
             <property name="minimumSize">
              <size>
               <width>400</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>700</width>
               <height>16777215</height>
              </size>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>Enter Password:</string>
                </property>
                <property name="buddy">
                 <cstring>editPassword</cstring>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QHBoxLayout" name="passwordLayout">
                <item>
                 <widget class="PasswordEdit" name="editPassword">
                  <property name="accessibleName">
                   <string>Password field</string>
                  </property>
                  <property name="echoMode">
                   <enum>QLineEdit::Password</enum>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QToolButton" name="buttonTogglePassword">
                  <property name="toolTip">
                   <string>Toggle password visibility</string>
                  </property>
                  <property name="accessibleName">
                   <string>Toggle password visibility</string>
                  </property>
                  <property name="checkable">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <spacer name="verticalSpacer_4">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeType">
                 <enum>QSizePolicy::Fixed</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>5</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>Enter Additional Credentials (if any):</string>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_5">
                <property name="sizeConstraint">
                 <enum>QLayout::SetMinimumSize</enum>
                </property>
                <item>
                 <spacer name="horizontalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeType">
                   <enum>QSizePolicy::Fixed</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>15</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <layout class="QGridLayout" name="gridLayout_3">
                  <property name="sizeConstraint">
                   <enum>QLayout::SetMinimumSize</enum>
                  </property>
                  <property name="topMargin">
                   <number>3</number>
                  </property>
                  <item row="1" column="3">
                   <layout class="QGridLayout" name="gridLayout">
                    <property name="spacing">
                     <number>0</number>
                    </property>
                    <item row="1" column="2">
                     <widget class="QProgressBar" name="yubikeyProgress">
                      <property name="maximumSize">
                       <size>
                        <width>16777215</width>
                        <height>2</height>
                       </size>
                      </property>
                      <property name="minimum">
                       <number>0</number>
                      </property>
                      <property name="maximum">
                       <number>0</number>
                      </property>
                      <property name="value">
                       <number>-1</number>
                      </property>
                      <property name="textVisible">
                       <bool>false</bool>
                      </property>
                     </widget>
                    </item>
                    <item row="0" column="2">
                     <widget class="QComboBox" name="comboChallengeResponse">
                      <property name="enabled">
                       <bool>false</bool>
                      </property>
                      <property name="sizePolicy">
                       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                        <horstretch>0</horstretch>
                        <verstretch>0</verstretch>
                       </sizepolicy>
                      </property>
                      <property name="accessibleName">
                       <string>Hardware key slot selection</string>
                      </property>
                      <property name="editable">
                       <bool>false</bool>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </item>
                  <item row="0" column="4">
                   <widget class="QPushButton" name="buttonBrowseFile">
                    <property name="toolTip">
                     <string>Browse for key file</string>
                    </property>
                    <property name="accessibleName">
                     <string>Browse for key file</string>
                    </property>
                    <property name="text">
                     <string>Browse...</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="3">
                   <layout class="QGridLayout" name="gridLayout_2">
                    <property name="verticalSpacing">
                     <number>0</number>
                    </property>
                    <item row="0" column="1">
                     <widget class="QComboBox" name="comboKeyFile">
                      <property name="enabled">
                       <bool>true</bool>
                      </property>
                      <property name="sizePolicy">
                       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                        <horstretch>0</horstretch>
                        <verstretch>0</verstretch>
                       </sizepolicy>
                      </property>
                      <property name="accessibleName">
                       <string>Key file selection</string>
                      </property>
                      <property name="editable">
                       <bool>true</bool>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </item>
                  <item row="1" column="0">
                   <layout class="QHBoxLayout" name="horizontalLayout_6">
                    <property name="spacing">
                     <number>5</number>
                    </property>
                    <item>
                     <widget class="QLabel" name="hardwareKeyLabel">
                      <property name="text">
                       <string>Hardware Key:</string>
                      </property>
                      <property name="buddy">
                       <cstring>comboChallengeResponse</cstring>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QToolButton" name="hardwareKeyLabelHelp">
                      <property name="cursor">
                       <cursorShape>PointingHandCursor</cursorShape>
                      </property>
                      <property name="toolTip">
                       <string>&lt;p&gt;You can use a hardware security key such as a &lt;strong&gt;YubiKey&lt;/strong&gt; or &lt;strong&gt;OnlyKey&lt;/strong&gt; with slots configured for HMAC-SHA1.&lt;/p&gt;
&lt;p&gt;Click for more information...&lt;/p&gt;</string>
                      </property>
                      <property name="accessibleName">
                       <string>Hardware key help</string>
                      </property>
                      <property name="styleSheet">
                       <string notr="true">QToolButton {
	border: none;
	background: none;
}</string>
                      </property>
                      <property name="text">
                       <string notr="true">?</string>
                      </property>
                      <property name="iconSize">
                       <size>
                        <width>12</width>
                        <height>12</height>
                       </size>
                      </property>
                      <property name="popupMode">
                       <enum>QToolButton::InstantPopup</enum>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </item>
                  <item row="0" column="0">
                   <layout class="QHBoxLayout" name="horizontalLayout_4">
                    <property name="spacing">
                     <number>5</number>
                    </property>
                    <item>
                     <widget class="QLabel" name="keyFileLabel">
                      <property name="text">
                       <string>Key File:</string>
                      </property>
                      <property name="buddy">
                       <cstring>comboKeyFile</cstring>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QToolButton" name="keyFileLabelHelp">
                      <property name="cursor">
                       <cursorShape>PointingHandCursor</cursorShape>
                      </property>
                      <property name="toolTip">
                       <string>&lt;p&gt;In addition to your master password, you can use a secret file to enhance the security of your database. Such a file can be generated in your database's security settings.&lt;/p&gt;&lt;p&gt;This is &lt;strong&gt;not&lt;/strong&gt; your *.kdbx database file!&lt;br&gt;If you do not have a key file, leave the field empty.&lt;/p&gt;&lt;p&gt;Click for more information...&lt;/p&gt;</string>
                      </property>
                      <property name="accessibleName">
                       <string>Key file help</string>
                      </property>
                      <property name="styleSheet">
                       <string notr="true">QToolButton {
	border: none;
	background: none;
}</string>
                      </property>
                      <property name="text">
                       <string>?</string>
                      </property>
                      <property name="iconSize">
                       <size>
                        <width>12</width>
                        <height>12</height>
                       </size>
                      </property>
                      <property name="popupMode">
                       <enum>QToolButton::InstantPopup</enum>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </item>
                  <item row="1" column="4">
                   <widget class="QPushButton" name="buttonRedetectYubikey">
                    <property name="enabled">
                     <bool>true</bool>
                    </property>
                    <property name="toolTip">
                     <string>Refresh hardware tokens</string>
                    </property>
                    <property name="accessibleName">
                     <string>Refresh hardware tokens</string>
                    </property>
                    <property name="text">
                     <string>Refresh</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QWidget" name="touchIDContainer" native="true">
                <layout class="QHBoxLayout" name="touchIDLayout">
                 <property name="spacing">
                  <number>0</number>
                 </property>
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>0</number>
                 </property>
                 <item>
                  <spacer name="horizontalSpacer">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>0</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item>
                  <widget class="QCheckBox" name="checkTouchID">
                   <property name="text">
                    <string>TouchID for Quick Unlock</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <layout class="QHBoxLayout" name="dialogButtonsLayout">
                <property name="topMargin">
                 <number>15</number>
                </property>
                <item alignment="Qt::AlignRight">
                 <widget class="QDialogButtonBox" name="buttonBox">
                  <property name="standardButtons">
                   <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_4">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::MinimumExpanding</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::MinimumExpanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>55</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
  <action name="keyFileClearIcon">
   <property name="text">
    <string>Clear</string>
   </property>
   <property name="toolTip">
    <string>Clear Key File</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PasswordEdit</class>
   <extends>QLineEdit</extends>
   <header>gui/PasswordEdit.h</header>
  </customwidget>
  <customwidget>
   <class>MessageWidget</class>
   <extends>QWidget</extends>
   <header>gui/MessageWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ElidedLabel</class>
   <extends>QLabel</extends>
   <header>gui/widgets/ElidedLabel.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>editPassword</tabstop>
  <tabstop>buttonTogglePassword</tabstop>
  <tabstop>comboKeyFile</tabstop>
  <tabstop>buttonBrowseFile</tabstop>
  <tabstop>hardwareKeyLabelHelp</tabstop>
  <tabstop>comboChallengeResponse</tabstop>
  <tabstop>buttonRedetectYubikey</tabstop>
  <tabstop>checkTouchID</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
